### ISSUE TITLE
Sprint - Jan 22 to Feb 2, 2024

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#19804]
# Global Sprint Planning

## 3 things that might take us down

1. Celery
2. Tim skiing
3. CH/infra being on fire while people are offline, sick and/or leave the company
4. Accidentally dropping data for billing migrations and with billing limits

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team Product Analytics

**Support hero:** 
- Michael

## Retro

- HogQL insights rewrite
   - All started, ongoing or close-to-completion... except for Funnels
- Support backlog
   - Still a backlog.
- Helping Pipeline with their new UI
   - There's a new UI (flagged?)
- Helping Pipeline with getting people off the last apps.
   - We did an analysis and [have a plan](https://github.com/PostHog/meta/issues/153#issuecomment-1894251008)

## OKR Q1 2024

1. HogQL & Data Exploration (Julian, Marius)
   - Convert ALL our insights to use HogQL as their base. **ongoing**
   - Improve the type system and get rid of `assumeNotNull`.
   - Write great docs for HogQL and data exploration nodes. **ongoing**
   - Proactively monitor query performance.

2. Product Analytics frontend (Thomas, Michael)
   - Clean things up now that we have PostHog 3000 and HogQL everywhere.
   - Work on the universal "Explore" view. 

3. BI (Tom)
   - Support non-event data sources on HogQL insights.
   - Build the new querying experience, and/or integrate it into the "Explore" view.

<details>
<summary>Action plan</summary>
- Come January, we'll take [this list](https://github.com/PostHog/meta/issues/130), split the checkboxes between the team, and keep going till it's done. ETA: end of January.
- In February (leap year, woohoo), we'll make Product Analytics ~great~ spark joy again: insights UI cleanup, unified data exploration experience, adding missing analysis features (lifecycle breakdowns, etc), and in parallel work on the BI and the HogQL editing experience. What _exactly_ we'll do and in what order is still TBD.
- In March we'll keep going. The goal is to have an amazing BI / Product Analytics / Insights / Explore / HogQL experience by the end of it.
</details>

## Next sprint

### High priority

- TODO

### Low priority / side quests

- TODO

**Comment by pauldambra** [comment]
Team Replay
Support hero: @marandaneto (@pauldambra supporting)

everybody is in the whole sprint 🚀 

## Retro
- very pleased that we're getting beta testers @pauldambra @marandaneto 
- wish there were fewer bug reports - support has been full on @pauldambra 

## Hang over items from previous sprint

✅ make sure the plans and controls change lands well with users @pauldambra
  - quite a lot of load from people wanting ingestion controls on old plan, but conversion to paid is roughly the same
✅  2 users using and have given us feedback on Android replay 
   - several users, not massive feedback yet, but getting there
✅  mobile dashboard template released 
  - PR waiting to land and then we can review and release this

## OKR

🟡📱 happy android replay users
  - feedback from beta testers but not a working recording from them yet 😓 
🟡 people love replay even more
  - still more bug reports than we'd like
🟢  first class mobile support
  - not focussed here, focus is on replay
  - but making small pieces of progress

## High priority

- network payload capture released for all @pauldambra
- 2 users using Android replay @marandaneto
- BETA canvas recording for replay @daibhin 

## Side quests

- AI session summary @pauldambra 
- (Wrap up comments work) - @benjackwhite and @daibhin needs some follow-up.
- POC of session replay with jetpack compose @marandaneto
- @daibhin take over this PR to add resizeable inspector tab for the player https://github.com/PostHog/posthog/pull/18378
- initial POC/discovery on React Native recordings

**Comment by robbie-c** [comment]
# Team Web Analytics

## Hang over items from previous sprint
* n/a

## OKR


### High priority
* Performance - spike of sampling in web analytics

### Low priority
* Other performance work
  * serialize queries on the page
  * turn on sampling for trends queries
  * think more about https://github.com/PostHog/posthog/issues/19289

**Comment by benjackwhite** [comment]
# Team Infra

**Support hero:** @frankh then @danielxnj 

## Retro
 

## Hang over items from previous sprint

1. New endpoints for ingestion / static assets
  * Still in progress due to Cloudflare trickiness 
2. Celery worker deployment
3. Testing of Contour / Envoy for ingress routing
4. IaC for Clickhouse EU

## OKR

1. 💪 Deploy with confidence
1. 💰 Keep costs under control
1. 👀 More Observability
1. 🎯 Soc2

### High priority

1. Rollout new ingestion endpoints and static asset accessing @frankh 
2. Scoped investigation into async django to relieve the concurrent requests issue @frankh @benjackwhite 
3. Scoped investigation into Soc2 (type I or II) with Fraser @danielxnj 

### Low priority / side quests
- Test out one of our Canary Deploy ideas
- RFC for the reverse proxy as a service @benjackwhite

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>
<summary>Retro items</summary>

  - [ ] Full testing of the onboarding flow @raquelmsmith 
  - [x] New session replay plan
    - https://github.com/PostHog/product-internal/pull/547
    - New plan and price
    - Feature gating based on available_product_features (right now it's behind a flag)
  - [x] Data pipelines pricing https://github.com/PostHog/billing/issues/417
    - https://github.com/PostHog/product-internal/pull/533/files
    - [x] Unsubscribe modal https://github.com/PostHog/posthog/pull/19334
    - [x] Create product in billing
    - [x] Add feature gates
    - [x] Organize launch plan
- [ ] Add metadata to startup plan customers in stripe @raquelmsmith - did not get to this yet, but we should
- [x] ctas on billing for customers over 1k/mo @xrdt https://github.com/PostHog/posthog/pull/19508
  - Bundling discounts
  - Team training
  - Roadmap chit chat 
- [ ] We shouldn’t show pricing above $20k/yr on the pricing pages so we don’t “shoot ourselves in the foot” for larger customers who will be paying above that
  - Could be an experiment, does it put people off from signing up because it isn’t as transparent, or feels enterprise-y. 
</details>


## OKR
- Double activation rates for product analytics and session replay
  - Polish onboarding flow with Cory
  - Run more smaller experiments
  - Do an average of 2 customer interviews per week
- Launch pricing for Data warehouse, queries, CDP


## This sprint
### High priority
- [ ] Full testing of the onboarding flow @raquelmsmith 
- [ ] Fix mrr per product for credit-based annual customers - @raquelmsmith 
- [x] Hide replay controls for people on the old plan - @xrdt 
- [ ] New product intro pages - https://github.com/PostHog/posthog/pull/19609 - @raquelmsmith 
- [x] Wrap up finished experiments - remove code, close experiment @xrdt 
  - [x] invite members https://us.posthog.com/project/2/experiments/8176
  - [x] post-ingestion redirect to insights https://us.posthog.com/project/2/experiments/8865

### Low priority / side quests
## Growth review items
- [ ] Add metadata to startup plan customers in stripe @raquelmsmith 
- [ ] We shouldn’t show pricing above $20k/yr on the pricing pages so we don’t “shoot ourselves in the foot” for larger customers who will be paying above that - @xrdt 
  - Could be an experiment, does it put people off from signing up because it isn’t as transparent, or feels enterprise-y. 
- [x] Make doc for how to create the right kinds of prices in Stripe @raquelmsmith 
- [x] Experiment: redirect people to web analytics after onboarding instead of insights screen - @xrdt

**Comment by EDsCODE** [comment]
# Team Data Warehouse

### OKR Q1 2024

### Objective

**Get EU moved over to new cluster**

* **Key Results:**
    * Terraform, ansible, data migrated over, backups, housewatch
    * Deprecate old cluster

**ByConity evaluation**

* **Key Results:**
    * Load up with all clickhouse data and benchmark

**5 reference customers with qualities:**

* **Key Results:**
    * Paying a reasonable amount
    * Tell us they’re delighted
    * Not a big edge case
    * Using our in house data import pipeline
    * Weekly queries on a data warehouse connected table
    * Saving queried tables to insights?

**Data Warehouse queries are compatible with insights hogQL**

 

### Retro

**Product**
- [ ] onboard more users to data connectors + release hubspot integration to interested users
- [ ] continue working through conversion from billing queries on metabase to within PostHog


**Infra** [Board](https://github.com/orgs/PostHog/projects/92/views/1)
- [ ] https://github.com/PostHog/posthog/issues/19420
- [ ] https://github.com/PostHog/posthog/issues/19421
- [ ] https://github.com/PostHog/posthog/issues/19422
- [ ] https://github.com/PostHog/posthog/issues/19423

### High priority

**Product**


**Infra** [Board](https://github.com/orgs/PostHog/projects/92/views/1)

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @liyiy (surveys) + @neilkakkar  
Days off:
Juraj: 1 days
Neil: 1 days
Li: 1 days

## Retro

- Release scheduled flag changes, fix some experiment bugs to get comfortable with the codebase to prep for Q1 OKRs ( #18104 , #18958 ) - @jurajmajerik --> Done, remaining bug will be done by Friday
- Release relative date targeting & targeting by distinct id - @neilkakkar --> Will release by friday
- Convert surveys to preact, and use it for previews in posthog app - @liyiy --> Bundling took unexpectedly long, will rollover to next sprint.


## Hang over items from previous sprint

---

## OKRs

1. Doubling down on surveys
2. Doubling down on experiments

### High priority

- e2e tests for experiments - @jurajmajerik 
- Experiment bug fixes, Common sense boundaries for predictions, update predictions after updating metrics - @jurajmajerik 
-  Statistical significance on secondary metrics & display secondary metrics charts - @neilkakkar 
- Finish converting surveys to preact, and use it for previews in posthog app - @liyiy
- Start on event based targeting implementation (RFC here: https://github.com/PostHog/product-internal/pull/556/ )- @liyiy 

### Low priority / side quests

- Temporal queues for feature success - @neilkakkar

**Comment by tiina303** [comment]
# Team Pipeline

- Offsite 22-26. Jan
- OOO: Tiina 29-31. Jan
- Support: Marcus, secondaries: Brett & Ted

## Retro

- [ ] Finish webhooks
    - [x] Delivery system testing in dev (Brett, Tomas)
    - [x] Deployment and monitoring (Xavier)
    - [x] Make plugin-server delegate to hooks (Xavier->Brett)
- [ ] Pipeline UI (Tiina)
    - [x] data pipelines unsubscribe option
    - [ ] new UI released
        - WIP but not yet released, needed to pick up support as Marcus was off 
- [ ] Finish POE (Ted)
    - [x] Write RFC for persons splits
    - [x] Move forward with backfill plan
        - triggered from the splits write-up we now have a simplified model for PoE distinct-id -> person-id overrides
    - [x] Continuing write ramp up

## Hang over items from previous sprint



- Webhooks & PoE & Pipeline UI will all be continued after the offsite.

## OKR 🟢 🔴 🟡 ⚪ 

- 🟡 Wrap up Q4 work before offsite in Jan
    - 🟡 Finish webhooks
        - 🟡 Delivery system (Brett, Tomás)
        - 🟡 Pipeline UI, apps hookup (Tiina)
        - Lots of progress was made for both, but not finished
    - 🟡 Finish POE (Ted)
        - Lots of progress, but not finished. Plan has been updated for a simpler solution 
    - 🟢 Finish capture-rs (Xavier)
         - backends depend on infra ingress work planned for Q1
- ⚪  Person model doesn't slow down the pipeline + doesn't live in shared PG (Dave)
- ⚪  Joint goal with replay team to get blobby stable (Dave)
    - ⚪  Determine demarc lines of ownership (API layer) going forward
    - ⚪  Make our part stable


### High priority

- [ ] Offsite (Tiina) 
    - [ ] RFC for the future of Person Model
    - [ ] Written plan for Blobby
- [ ] Webhooks delivery system running idle in US and EU cloud (Xavier)
- [ ] PoE: Write path for the new distinctId->PersonId overrides (Ted)

### Low priority / side quests

- [ ] HogQL batch exports events filtering (Tomas)
